controllers/configurationpolicy_controller.go

```diff
@@ -1187,14 +1190,17 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(plc policyv1.Confi
 		}
 	}
 
-	r.checkRelatedAndUpdate(plc, relatedObjects, oldRelated, parentStatusUpdateNeeded)
+	r.checkRelatedAndUpdate(plc, relatedObjects, oldRelated, parentStatusUpdateNeeded, false)
```

Comment:
I think this one should be `true`, I think this is the "happy path" and that `true` is be needed here to ensure that when the related objects changed, then old ones are correctly removed... this might be the cause of the currently failing test in the PR.